src/meta/MetaServiceUtils.cpp

```diff
@@ -295,11 +310,49 @@ std::string MetaServiceUtils::leaderVal(const LeaderParts& leaderParts) {
     return value;
 }
 
+// v3: dataVer(int) + lenOfHost(8) + HostAddr(varchar) + term(int64_t)
+std::string MetaServiceUtils::leaderValV3(const HostAddr& h, int64_t term) {
+    std::string leaderVal;
+    leaderVal.reserve(256);
+
+    int dataVersion = 3;
+
+    auto sHost = serializeHostAddr(h);
+    auto lenOfHost = sHost.size();
+    LOG(INFO) << "lenOfHost = " << lenOfHost;
+
+    leaderVal.append(reinterpret_cast<const char*>(&dataVersion), sizeof(dataVersion))
+             .append(reinterpret_cast<const char*>(&lenOfHost), sizeof(lenOfHost))
+             .append(sHost)
+             .append(reinterpret_cast<const char*>(&term), sizeof(term));
+    return leaderVal;
+}
+
+// v3: dataVer(int) + lenOfHost(8) + HostAddr(varchar) + term(int64_t)
+std::pair<HostAddr, int64_t> MetaServiceUtils::parseLeaderValV3(folly::StringPiece val) {
```

Comment:
Maybe we need to consider backward compatibility here, return a Status or sth else. And handle it in `ActiveHostsMan::updateHostInfo`